src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -66,71 +66,78 @@ public string DefaultConfigurationPath
+                        }
+                        ExecutionException = ex;
+                        return FAILURE;
+                    }
+                    finally
                     {
-                        // These exceptions escaped our net and must be logged here
-                        RuntimeErrors |= Errors.LogUnhandledEngineException(_rootContext, ex);
+                        logger.AnalysisStopped(RuntimeErrors);
                     }
-                    ExecutionException = ex;
-                    return FAILURE;
                 }
-                finally
-                {
-                    logger.AnalysisStopped(RuntimeErrors);
-                }
-            }
 
-            bool succeeded = (RuntimeErrors & ~RuntimeConditions.Nonfatal) == RuntimeConditions.None;
+                bool succeeded = (RuntimeErrors & ~RuntimeConditions.Nonfatal) == RuntimeConditions.None;
 
-            if (succeeded)
-            {
-                try
+                if (succeeded)
                 {
-                    ProcessBaseline(_rootContext, options, FileSystem);
-                }
-                catch (Exception ex)
-                {
-                    RuntimeErrors |= RuntimeConditions.ExceptionProcessingBaseline;
-                    ExecutionException = ex;
-                    return FAILURE;
-                }
+                    try
+                    {
+                        ProcessBaseline(_rootContext, options, FileSystem);
+                    }
+                    catch (Exception ex)
+                    {
+                        RuntimeErrors |= RuntimeConditions.ExceptionProcessingBaseline;
+                        ExecutionException = ex;
+                        return FAILURE;
+                    }
 
-                try
-                {
-                    PostLogFile(options.PostUri, options.OutputFilePath, FileSystem);
+                    try
+                    {
+                        PostLogFile(options.PostUri, options.OutputFilePath, FileSystem);
+                    }
+                    catch (Exception ex)
+                    {
+                        RuntimeErrors |= RuntimeConditions.ExceptionPostingLogFile;
+                        ExecutionException = ex;
+                        return FAILURE;
+                    }
                 }
-                catch (Exception ex)
+
+                if (options.RichReturnCode)
                 {
-                    RuntimeErrors |= RuntimeConditions.ExceptionPostingLogFile;
-                    ExecutionException = ex;
-                    return FAILURE;
+                    return (int)RuntimeErrors;
                 }
-            }
 
-            if (options.RichReturnCode)
+                return succeeded ? SUCCESS : FAILURE;
+            }
+            finally
             {
-                return (int)RuntimeErrors;
+                _rootContext?.Dispose();
```

Comment:
>_rootContext?.Dispose [](http://example.com/codeflow?start=16&length=21)

Here's the key change. In all code paths on calling Run, we dispose the root context object if non-null.